src/auth/authenticator/tls/authenticator.go

```diff
@@ -52,7 +52,8 @@ func (auth Authenticator) Authenticate(req *http.Request) (*types.UserInfo, erro
 		AuthMode: AuthMode,
 	}
 	userInfo.Username, userInfo.Tenant = utils.ExtractUsernameAndTenant(clientCert.Subject.CommonName)
-	userInfo.Roles, userInfo.Permissions = utils.ExtractRolesAndPermission(clientCert.Subject.Organization)
+	userInfo.Permissions = utils.ExtractPermissions(clientCert.Subject.OrganizationalUnit)
+	userInfo.Roles = clientCert.Subject.Organization
```

Comment:
I would rather have a utility function for it

Reply:
Is it necessary ? We assign an array to another one here...

Reply:
It is not very neccesary but I think it believe it would be good to have a single share method for every auth middleware to extract roles out of string or array. But this is optional